alacritty/src/renderer/mod.rs

```diff
@@ -268,8 +272,12 @@ impl GlyphCache {
 
         // Rasterize the glyph using the built-in font for special characters or the user's font
         // for everything else.
-        let rasterized = builtin_font::builtin_glyph(glyph_key.character, &self.metrics)
-            .map_or_else(|| self.rasterizer.get_glyph(glyph_key), Ok);
+        let rasterized = if self.builtin_box_drawing {
+            builtin_font::builtin_glyph(glyph_key.character, &self.metrics)
+                .map_or_else(|| self.rasterizer.get_glyph(glyph_key), Ok)
+        } else {
+            self.rasterizer.get_glyph(glyph_key)
+        };
```

Comment:
I couldn't figure out how to make this work with `.then()`.

Reply:
Try the following:

```diff
diff --git a/alacritty/src/renderer/mod.rs b/alacritty/src/renderer/mod.rs
index a3184d1c..3d7ecc74 100644
--- a/alacritty/src/renderer/mod.rs
+++ b/alacritty/src/renderer/mod.rs
@@ -272,12 +272,11 @@ impl GlyphCache {

         // Rasterize the glyph using the built-in font for special characters or the user's font
         // for everything else.
-        let rasterized = if self.builtin_box_drawing {
-            builtin_font::builtin_glyph(glyph_key.character, &self.metrics)
-                .map_or_else(|| self.rasterizer.get_glyph(glyph_key), Ok)
-        } else {
-            self.rasterizer.get_glyph(glyph_key)
-        };
+        let rasterized = self
+            .builtin_box_drawing
+            .then(|| builtin_font::builtin_glyph(glyph_key.character, &self.metrics))
+            .flatten()
+            .map_or_else(|| self.rasterizer.get_glyph(glyph_key), Ok);

         let glyph = match rasterized {
             Ok(rasterized) => self.load_glyph(loader, rasterized),
```